main.py

```diff
@@ -38,6 +38,11 @@ def main():
         match option:
             case "0":
                 df = import_a_csv_file()
+            case "3":
+                try:
+                    data_management.add_transaction(df)
```

Comment:
```suggestion
                    df = data_management.add_transaction(df)
```

You current code does not affect `df`.
You need to overwrite the valuable `df` by assigning return value of `data_management.add_transaction(df)`.